core/integration/container_test.go

```diff
@@ -754,10 +919,10 @@ func TestContainerImageExport(t *testing.T) {
 		)
 		require.NoError(t, err)
 
-		testRef := "127.0.0.1:5000/testimagepush:latest"
+		testRef := core.ContainerAddress("127.0.0.1:5000/testimagepush:latest")
```

Comment:
nit (on the spec, not the implementation): I think this should be a string rather than a scalar. Scalars define new types and are mostly useful for things that are "passthrough" to the client (e.g. an `ID`, which the user is supposed to retrieve and pass to another query, not create manually).

Container addresses however are "manually" entered by the user and I think having a scalar is not helping (and the user needs to do casting in their code to use it)

Reply:
I'm inclined to agree. Opened https://github.com/dagger/dagger/issues/3257 for reaching consensus on this.